riscv/csrs.cc

```diff
@@ -1460,6 +1462,17 @@ void sstateen_csr_t::verify_permissions(insn_t insn, bool write) const {
       throw trap_virtual_instruction(insn.bits());
 }
 
+// implement class menvcfg_csr_t
+menvcfg_csr_t::menvcfg_csr_t(processor_t* const proc, const reg_t addr, const reg_t mask,
+                             const reg_t init):
+  masked_csr_t(proc, addr, mask, init) {
+}
+
+bool menvcfg_csr_t::unlogged_write(const reg_t val) noexcept {
+  reg_t adjusted_val = get_field(val, MENVCFG_CBIE) != CBIE_RESERVED ? val : set_field(val, MENVCFG_CBIE, 0);
+  return masked_csr_t::unlogged_write(adjusted_val);
+}
```

Comment:
It's pretty ugly to have the same WARLization in three separate places here.

Since CBIE is the same for all three of these `*envcfg` CSRs, how about having `senvcfg` and `henvcfg` inheriting from a parent class that does the WARLization? In that parent class's constructor you can `assert(MENVCFG_CBIE == SENVCFG_CBIE && MENVCFG_CBIE == HENVCFG_CBIE);` to make sure the one implementation will work properly for all three CSRs.

Reply:
Add ```envcfg_csr_t``` as the parent class that does the WARLization and assertion.